auslib/test/admin/views/test_rules.py

```diff
@@ -141,9 +141,22 @@ def testNewRuleWithoutPermission(self):
     def testMissingFields(self):
         # But we still need to pass product, because permission checking
         # is done before what we're testing
-        ret = self._post('/rules', data=dict({'product': 'a'}))
+        post_data = {'product': 'a'}
+        ret = self._post('/rules', data=post_data)
+        self.assertEquals(ret.status_code, 400, "Status Code: %d, Data: %s" % (ret.status_code, ret.data))
+        # Connexion and manual request.json validation methods will preempt and throw 400 error as
+        # soon as the first field invalidates. Only one field may be present in the response.data depending
+        # upon the order of fields sorted in swagger yaml file and in views.
+        self.assertTrue('title' in ret.data, msg=ret.data)
```

Comment:
That is a bit more trickier to manage. This response is only seen in test cases or when we attempt to invoke the API apart from the UI page , eg. a curl request.
When I try to submit a new rule from UI page I get [this](https://pastebin.mozilla.org/9020893) in the logs. 

Two observations from it, UI forms send a blank string value i.e. '' instead of null or none to the server via the wtfForms. Connexion acting in middle-ware for our flask app  intercepts the request automatically, validates the values based on the swagger spec file and immediately throws error. However, it fails to specify the field name which was invalid and neither does it highlight the invalid fields on UI page.